pkg/brokerapi/openservicebroker/open_service_broker_client.go

```diff
@@ -186,3 +210,39 @@ func (c *openServiceBrokerClient) DeleteServiceBinding(sID, bID string) error {
 
 	return nil
 }
+
+func (c *openServiceBrokerClient) pollBroker(ID string, operation string) error {
+	pollReq := brokerapi.LastOperationRequest{}
+	if operation != "" {
+		pollReq.Operation = operation
+	}
+
+	url := fmt.Sprintf(pollingFormatString, c.broker.Spec.URL, ID)
+	for i := 0; i < pollingAmountLimit; i++ {
+		glog.Infof("Polling attempt #%v: %s", i+1, url)
+		pollResp, err := util.SendRequest(c.client, http.MethodGet, url, pollReq)
+		if err != nil {
+			return err
+		}
+
+		lo := brokerapi.LastOperation{}
+		if err := util.ResponseBodyToObject(pollResp, &lo); err != nil {
+			return err
+		}
+
+		switch lo.State {
+		case "in progress":
+		case "succeeded":
+			return nil
+		case "failed":
```

Comment:
Yes, they're strings passed back in the state attribute of the response. The only possible status codes during the polling loop are 200 OK and 410 Gone. Here's where it talks about the async stuff: [OSB API](https://docs.cloudfoundry.org/services/api.html#async)